### PR TITLE
Memory Controller (MC) Error Detection And Correction (EDAC)

### PR DESCRIPTION
**Pull request by ktsaou**
- [x] rewrite of the memory controller edac collector
- [x] one chart per memory controller
- [x] collect errors without info too
- [ ] support a chart per dimm
- [ ] update the alarms to use the new charts

### CODE REVIEW

**Comment by ktsaou** [comment]
@ilyam8 I have not tested this. Please test it on mega and beast.
Also please help finish it (alerts, docs, etc)

File: collectors/proc.plugin/sys_devices_system_edac_mc.c [hunk 1]
```diff
@@ -42,162 +58,233 @@ static void find_all_mc() {
...
+            continue;
+        }
 
-    if(unlikely(mc_root == NULL)) {
-        find_all_mc();
-        if(unlikely(mc_root == NULL))
-            return 1;
-    }
+        while((de = readdir(dir))) {
+            if(de->d_type == DT_DIR && strncmp(de->d_name, "rank", 4) == 0 && isdigit(de->d_name[4])) {
```
Comment (ilyam8) [inline collectors/proc.plugin/sys_devices_system_edac_mc.c:96]: At this level, it could be either the `dimm` or `rank` directory, see "[dimmX or rankX directories](https://www.kernel.org/doc/html/v5.0/admin-guide/ras.html#dimmx-or-rankx-directories)"

TL;DR

> The recommended way to use the EDAC subsystem is to look at the information provided by the **dimmX** or **rankX** directories.
>
> On some systems, the memory controller doesn’t have any logic to identify the memory module. On such systems, the directory is called **rankX**.
>
> On modern Intel memory controllers, the memory controller identifies the memory modules directly. On such systems, the directory is called **dimmX**.

**Comment by ilyam8** [comment]
> update documentation

The current proc.plugin readme has no mention of EDAC, I think we will have it after generating docs from metadata, so no need to update now.
